pages/index.js

```diff
@@ -23,6 +25,26 @@ class Index extends Component {
   componentDidMount() {
     connectSocket()
     this.props.dispatch(projectsListener())
+    this.updateStateFromQueries()
+  }
+
+  componentDidUpdate(previousProps) {
```

Comment:
I think you want `componentWillReceiveProps(newProps)` although perhaps it doesn't matter. That's just the one I'm used to using.

Reply:
I also thought this, but the docs say that `componentWillReceiveProps` is legacy 

https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops